### ISSUE TITLE
ttypescript does not look for typescript or transformers in expected places

### ISSUE THREAD

**Issue by dgoldstein0** [cevek/ttypescript#57]
I am attempting to run ttypescript with my company's build system, and we have a slightly nonstandard way of using node; we expect ttypescript to search NODE_PATH, like node does when it's set, to find transforms & for typescript; we also expect it to honor --preserve-symlinks.

I'm not totally sure why it's not working, but I noticed that ttypescript uses resolve(), and almost certainly uses different settings for it than what I'm invoking node with, which makes it not work.  Depending on how I am running my ttypescript binary, either I'm getting it crashing trying to load the transformer, or when trying to load typescript.

**Comment by cevek** [comment]
Can you provide some example project?

**Comment by dgoldstein0** [comment]
unfortunately I was doing this with a huge amount of non open-source stuff.  But to summarize some of the highlights that are probably important
- we have a single node_modules/ folder, it's pointed to by NODE_PATH.  my typescript install is symlinked into here. as well as ttypescript
- we run node like `NODE_PATH=<our node_modules/> node --preserve-symlinks --preserve-symlinks-main symlink/to/main.js`.  I think main here was ttypescript iirc.
- every .js file is actually a symlink to the actual file.

so basically ttypescript needs to be invoked in some manner like that

At this point I'm not very interested in constructing a repro - it's been ~5 months, I don't really remember everything and I didn't save my work, and was able to get the transformer working at the time by cutting ttypescript out of the process

**Comment by cevek** [comment]
Ok, thank you for explanation. 
Your case is very specific, so you can as you said use typescript api to do more specific settings.
ttypescript is just like a sugar over typescript api.
Main goal of that is to reduce pain for a common scenarios.
